rfc/composite_table_availability.pseudocode

```diff
@@ -0,0 +1,123 @@
+
+PhysicalTableSchema:
+	getFieldName(Column c)
+
+DataSource:
+	String getName()
+	Map<FieldName, Set<Segment>> raw segment data
+	Map<FieldName, List<Interval>> squashedFromSegments  // optional optimization
```

Comment:
As a matter of practice, SimplifiedIntervalList should be used to coalesce intervals into a simpler form.  I'm not convinced it won't go away in favor of a more standard library, so I've been reluctant to put it into interfaces more than necessary.

SimplifiedIntervalList is actually a little costly to build, so I'd rather not force people to build them more often than necessary.  It is /advisable/ to simplify, but carrying simplified data around in a list should be reasonable.